### PR TITLE
Add iiif media type and mirador media element

### PR DESCRIPTION
**Pull request by dodget**
This PR adds the ability to import IIIF content into scalar by:
- Defining a IIIF mediatype
- Defining a Mirador mediaobject
- Including a minified build of Mirador3

Currently the way this works is to upload a manifest by its url via the "Link to Media Files" functionality.
In order for a manifest to be parsed, the url must have `?iiif-manifest=1`. This is currently a way to test functionality, but the queues to trigger the parsing of a manifest should be more robust before this functionality is merged.

### CODE REVIEW

File: system/application/views/widgets/mediaelement/jquery.mediaelement.js [hunk 1]
```diff
@@ -5488,6 +5504,63 @@ function YouTubeGetID(url){
...
+        ]
+      });
+
+			this.parentView.layoutMediaObject();
+			this.parentView.removeLoadingMessage();
+
+			return;
+		}
+
+		// These functions are basically irrelevant for this type of media
```
Comment (ColeDCrawford) [inline system/application/views/widgets/mediaelement/jquery.mediaelement.js:5545]: Do we need jQuery.MiradorObjectView.prototype.play and those other functions?

IIIF will eventually expand to include audio/video content, but not for quite some time. Not within our scope.

**Comment by ColeDCrawford** [comment]
This is looking good from the screenshot. We'll definitely need to create a stripped down instance of Mirador for this, but let's keep using the standard build for now.

File: system/application/views/widgets/mediaelement/jquery.mediaelement.js (hunk 1 above)
Comment (dodget) [reply system/application/views/widgets/mediaelement/jquery.mediaelement.js:5545]: I went back and rewatched the zoom call we had with Erik and Craig, and from what I understood those methods need to be present, but set to null or empty functions if you don't want the functionality. I modeled it on the Deep Zoom player that the've set up.

**Comment by ColeDCrawford** [comment]
- Looks like the top black bar in M3 is caused by a header selector in Scalar's [common.css](https://scalar-dev.fas.harvard.edu/system/application/views/melons/cantaloupe/css/common.css) which is setting some padding. Adding something like `.mirador-viewer header { padding:0px }` would fix it, just not sure where that should go.
- I see what you mean about text in Mirador being small. We can work on that, but I don't think it should hold up the PR.
- On the "Edit" page for a IIIF media item, "Is IIIF manifest" should probably be checked by default. What would the expected behavior be for unchecking this? Perhaps we just remove that option on the edit page - not sure if that's the exact same template as the `new.edit` page though. If you recheck it on the edit page and save, it appends an additional `?iiif-manifest=1`.
- We should also provide the option for uploading a IIIF manifest as a file. When a new IIIF media item is created by linking to a manifest, `?iiif-manifest=1` is appended to the URL. Is the media type being saved in the database or is that JS function parsing the URL every time? We'll probably need to append that param here too. Adding this option to the file upload page can be a new separate story.

**Review by ColeDCrawford — APPROVED** [review]

**Pull request merged**